app/javascript/components/admin/AuditModal.js

```diff
@@ -0,0 +1,338 @@
+      let all_jurisdiction_paths = Object.assign({}, ...Object.entries(responseData).map(([id, path]) => ({ [path]: parseInt(id) })));
+
+      this.setState({ all_jurisdiction_paths });
+    });
+  }
+
+  formatChange = change => {
+    switch (change.name) {
+      case 'locked_at':
+        if (!change.details || !Array.isArray(change.details) || !change.details.length) {
+          return (
+            <span>
+              <b>Account Status</b>: Updated
+            </span>
+          );
+        } else if (change.details[0]) {
+          return (
+            <span>
+              <b>Account Status</b>: Unlocked
+            </span>
+          );
+        } else {
+          return (
+            <span>
+              <b>Account Status</b>: Locked
+            </span>
+          );
+        }
+      case 'jurisdiction_id':
+        if (!change.details || !Array.isArray(change.details) || change.details.length < 2) {
+          return (
+            <span>
+              <b>Jurisdiction</b>: Updated
+            </span>
+          );
+        } else {
+          return (
+            <span>
+              <b>Jurisdiction</b>: Changed from &quot;{_.invert(this.state.all_jurisdiction_paths)[change.details[0]]}&quot; to &quot;
+              {_.invert(this.state.all_jurisdiction_paths)[change.details[1]]}&quot;
+            </span>
+          );
+        }
+      case 'created_at':
+        return (
+          <span>
+            <b>Account Created</b>
+          </span>
+        );
+      case 'api_enabled':
+        if (!change.details || !Array.isArray(change.details) || !change.details.length) {
+          return (
+            <span>
+              <b>API Access</b>: Updated
+            </span>
+          );
+        } else if (change.details[0]) {
+          return (
+            <span>
+              <b>API Access</b>: Disabled
+            </span>
+          );
+        } else {
+          return (
+            <span>
+              <b>API Access</b>: Enabled
+            </span>
+          );
+        }
+      case 'role':
+        if (!change.details || !Array.isArray(change.details) || change.details.length < 2) {
+          return (
+            <span>
+              <b>Role</b>: Updated
+            </span>
+          );
+        } else {
+          return (
+            <span>
+              <b>Role</b>: Changed from &quot;{change.details[0]}&quot; to &quot;{change.details[1]}&quot;
+            </span>
+          );
+        }
+      case 'email':
+        if (!change.details || !Array.isArray(change.details) || change.details.length < 2) {
+          return (
+            <span>
+              <b>Email</b>: Updated
+            </span>
+          );
+        } else {
+          return (
+            <span>
+              <b>Email</b>: Changed from &quot;{change.details[0]}&quot; to &quot;{change.details[1]}&quot;
+            </span>
+          );
+        }
+      case 'authy_enabled':
```

Comment:
When I reset 2FA for a user, I don't see it show up in the Audit history - so I'm assuming there may be an issue with this check since it looks like we are correctly auditing `authy_enabled` in `user.rb`.

Reply:
This is currently available on the demo site as is right now and I tested the 2FA reset and it seems to be working proprly there (you can check my rhack account on demo to see if you'd like). Could your local environament be behaving slightly differently which might cause you to not see this audit? The Enabled v. Disabled audits displayed are acutally flipped so I updated that.

Reply:
Yes sorry I thought and went back to adjust this comment - 2FA isn't enabled locally by default for users but I do see it on demo so all set there. Thanks for catching the other issue!